lib/publiccloud/gcp_client.pm

```diff
@@ -71,7 +72,27 @@ sub get_next_vault_role {
 sub create_credentials_file {
     my ($self) = @_;
     my $credentials_file;
+    if (get_var('PUBLIC_CLOUD_CREDENTIALS_URL')) {
+        my $json = get_credentials();
+        # Need to remove quotes, convert \n to \\n and / to \/
+        assert_script_run(q{export key=$(cat /root/credentials.json | jq '.private_key_data' | sed 's/\\"//g' | sed 's/\\\n/\\\\\\\n/g' | sed 's/\\//\\\\\//g')});
+        #assert_script_run(q{export key=$(cat /root/credentials.json | jq '.private_key_data' | sed 's/\"//g' | sed 's/\\n/\\\\n/g'      | sed 's/\//\\\//g')});
```

Comment:
What are you trying to do over here? Do you know `jq -r`?

Reply:
yes, I know, and that was the first approach I used, but it removed all the `\n` which we need in plain text for the credentials file. I found out yesterday that how Google treats credentials is very stupid...
For example:

```
...
  "private_key": "-----BEGIN PRIVATE KEY-----\nXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZ\n-----END PRIVATE KEY-----\n",
  "client_email": "some@email",
  ...
```
works, but
```
...
  "private_key": "-----BEGIN PRIVATE KEY-----
XYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZXYZ
-----END PRIVATE KEY-----
",
  "client_email": "some@email",
  ...
```
doesn't... 

So, we need to keep the `\n` as strings, and `jq -r` removes them.

Reply:
OK I see and agree. But see https://github.com/stedolan/jq/issues/1881#issuecomment-479518797